homeassistant/components/rest/switch.py

```diff
@@ -119,30 +119,30 @@ def __init__(
             unique_id=unique_id,
         )
 
-        auth = None
+        auth: aiohttp.BasicAuth | None = None
+        username: str | None = None
         if username := config.get(CONF_USERNAME):
-            auth = aiohttp.BasicAuth(username, password=config[CONF_PASSWORD])
-
-        self._resource = config.get(CONF_RESOURCE)
-        self._state_resource = config.get(CONF_STATE_RESOURCE) or self._resource
-        self._method = config.get(CONF_METHOD)
-        self._headers = config.get(CONF_HEADERS)
-        self._params = config.get(CONF_PARAMS)
+            password: str = config[CONF_PASSWORD]
+            auth = aiohttp.BasicAuth(username, password=password)
+
+        self._resource: str = config[CONF_RESOURCE]
+        self._state_resource: str = config.get(CONF_STATE_RESOURCE) or self._resource
+        self._method: str = config[CONF_METHOD]
+        self._headers: dict[str, template.Template] | None = config.get(CONF_HEADERS)
+        self._params: dict[str, template.Template] | None = config.get(CONF_PARAMS)
```

Comment:
General philosophical question regarding these type hints – `config` is just `dict[str, Any]`, so doesn't this kinda mean that you're just pinky-swearing that you know perfectly well what type you'll get from the `config` for each key, and the type-checker just best believe it?

It sounds like the chances for someone changing a schema (or the value type for a key, in the generic case) and not changing the type annotations elsewhere are awfully high.

Reply:
All type hints in python are just "pinky-swearing" anyway.
It is the job of the developper and the reviewer to ensure that the schema matches the type hints.

Type hints are extremely useful to help spot bad code, or identify gaps in the handling.

Reply:
Sure, but adding types _here_ is even more pinky-sweary instead of at the source (if such a thing was possible, e.g. with a `mypy-voluptuous` plugin that would understand how a schema maps into a dict).